##
 Checking the validity of Simple Linear regression assumptions using the Palmer Penguins dataset on Seaborn

In [1]:
# Importing the important packages
import pandas as pd
import seaborn as sns

In [2]:
# loading dataset 
penguins = sns.load_dataset("penguins")

#checking the first 5 items in the data set 
penguins.head()

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   

   body_mass_g     sex  
0       3750.0    Male  
1       3800.0  Female  
2       3250.0  Female  
3          NaN     NaN  
4       3450.0  Female

In [4]:
# checking how the data set looks like how many rows and rows 
penguins.shape

(344, 7)

There are 7 rows and 344 columns in the dataset 
The columns have variables such as 
    - species
    - island
    - bill_length_mm
    - bill_depth_mm
    - flipper_length_mm
    - body_mass_g
    - sex

There is also some missing data in the dataset as there are NaN Types so we will have to clean them 

In [6]:
#Checking how many types of penguin species there are in the species column
species_type = penguins['species'].unique()
print(species_type)

['Adelie' 'Chinstrap' 'Gentoo']


In [17]:
# now i know i have 3 types of penguin species but i don't want to use them all for this analysis
# i want to only use only "Adelie" and "Gentoo" 
#I also want to drop missing values 

penguins_sub = penguins[penguins["species"] != "Chinstrap"] #selecting needed columns
penguins_final = penguins_sub.dropna() #droping NaN from the selected columns
penguins_final.reset_index(inplace=True, drop=True)


In [ ]:
#Other ways to do the same thing
#using .isin
# penguins_sub = penguins[penguins["species"].isin(["Adelie", "Gentoo"])]

#using OR
# penguins_sub = penguins[(penguins["species"] == "Adelie") | (penguins["species"] == "Gentoo")]

